specifyweb/frontend/js_src/lib/components/Toolbar/Query.tsx

```diff
@@ -198,7 +198,9 @@ export function QueryListDialog({
         </table>
         <span className="-ml-2 flex-1" />
         {data === undefined && loadingGif}
-        {paginator(data?.totalCount)}
+        {data !== undefined && data?.records.length > 0
+          ? paginator(data?.totalCount)
```

Comment:
no need for `?` here since you asserted on a previous line that it's not undefined

```suggestion
          ? paginator(data.totalCount)
```